pkg/controller/certificates/trigger/policies/gatherer_test.go

```diff
@@ -0,0 +1,256 @@
+				{ObjectMeta: metav1.ObjectMeta{OwnerReferences: []metav1.OwnerReference{{UID: "uid-1", Controller: pointer.BoolPtr(true)}}}},
+			}, nil),
+			wantCurrentRevisionRequest: nil,
+		},
+		{
+			name: "should return the certificaterequest and secret and both found",
+			givenCrt: &cmapi.Certificate{
+				ObjectMeta: metav1.ObjectMeta{UID: "uid-1"},
+				Spec:       cmapi.CertificateSpec{SecretName: "secret-1"},
+				Status:     cmapi.CertificateStatus{Revision: ptr(1)},
+			},
+			mockSecretLister: mockSecretLister("default", "secret-1", &v1.Secret{ObjectMeta: metav1.ObjectMeta{Name: "secret-1"}}, nil),
+			mockListRequests: mockRequestLister("default", "", []*cmapi.CertificateRequest{
+				{ObjectMeta: metav1.ObjectMeta{
+					OwnerReferences: []metav1.OwnerReference{{UID: "uid-1", Controller: pointer.BoolPtr(true)}},
+					Annotations:     map[string]string{"cert-manager.io/certificate-revision": "1"}},
+				},
+			}, nil),
+			wantCurrentRevisionRequest: &cmapi.CertificateRequest{ObjectMeta: metav1.ObjectMeta{
+				OwnerReferences: []metav1.OwnerReference{{UID: "uid-1", Controller: pointer.BoolPtr(true)}},
+				Annotations:     map[string]string{"cert-manager.io/certificate-revision": "1"}},
+			},
+			wantSecret: &v1.Secret{ObjectMeta: metav1.ObjectMeta{Name: "secret-1"}},
+		},
+		{
+			name:             "should return error when multiple certificaterequests found",
+			givenCrt:         &cmapi.Certificate{ObjectMeta: metav1.ObjectMeta{UID: "uid-1"}, Status: cmapi.CertificateStatus{Revision: ptr(1)}},
+			mockSecretLister: mockSecretLister("default", "", nil, nil),
+			mockListRequests: mockRequestLister("default", "", []*cmapi.CertificateRequest{
+				{ObjectMeta: metav1.ObjectMeta{OwnerReferences: []metav1.OwnerReference{{UID: "uid-1", Controller: pointer.BoolPtr(true)}}, Annotations: map[string]string{"cert-manager.io/certificate-revision": "1"}}},
+				{ObjectMeta: metav1.ObjectMeta{OwnerReferences: []metav1.OwnerReference{{UID: "uid-1", Controller: pointer.BoolPtr(true)}}, Annotations: map[string]string{"cert-manager.io/certificate-revision": "1"}}},
+			}, nil),
+			wantErr: "multiple CertificateRequest resources exist for the current revision, not triggering new issuance until requests have been cleaned up",
+		},
+		{
+			name:             "should return error when the list func returns an error",
+			givenCrt:         &cmapi.Certificate{ObjectMeta: metav1.ObjectMeta{UID: "uid-1"}, Status: cmapi.CertificateStatus{Revision: ptr(1)}},
+			mockSecretLister: mockSecretLister("default", "", nil, nil),
+			mockListRequests: mockRequestLister("default", "", []*cmapi.CertificateRequest{}, fmt.Errorf("some error from certificates.List that is not not_found")),
+			wantErr:          "some error from certificates.List that is not not_found",
+		},
+	}
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
```

Comment:
```go
for name, test := range tests {
  t.Run(name, func(t *testing.T) {
...
```

Reply:
Hmm i guess i'll need to tweak `gotests`, that's where this `tt` and `name := []struct` come from 😅